presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -94,31 +108,16 @@
                         "    smallint_to_bigint         SMALLINT," +
                         "    int_to_bigint              INT," +
                         "    bigint_to_varchar          BIGINT," +
-                        "    float_to_double            FLOAT" +
+                        format("    float_to_double            %s,", floatToDoubleType) +
+                        format("    row_to_row                 STRUCT < tinyint_to_smallint : TINYINT, tinyint_to_int : TINYINT, tinyint_to_bigint : TINYINT, smallint_to_int : SMALLINT, smallint_to_bigint : SMALLINT, int_to_bigint : INT, bigint_to_varchar : BIGINT, float_to_double : %s >,", floatToDoubleType) +
+                        format("    list_to_list               ARRAY < STRUCT < tinyint_to_smallint : TINYINT, tinyint_to_int : TINYINT, tinyint_to_bigint : TINYINT, smallint_to_int : SMALLINT, smallint_to_bigint : SMALLINT, int_to_bigint : INT, bigint_to_varchar : BIGINT, float_to_double : %s > >,", floatToDoubleType) +
+                        format("    map_to_map                 MAP < TINYINT, STRUCT < tinyint_to_smallint : TINYINT, tinyint_to_int : TINYINT, tinyint_to_bigint : TINYINT, smallint_to_int : SMALLINT, smallint_to_bigint : SMALLINT, int_to_bigint : INT, bigint_to_varchar : BIGINT, float_to_double : %s > >", floatToDoubleType) +
```

Comment:
Your test is very complete. However, I find it a bit too hard to read.

For readability, I make a few proposals:
* Use abbreviations. (We avoid abbreviation in Presto in general. But I find it excessively wordy in this case otherwise)
* Avoid unnecessary spaces.
* Don't cover all conversions in each of row_to_row, list_to_list, map_to_map. Instead, cover all of them across the three. (And add comment to point this out.)

In addition, I find it useful to have non coerced items in the structs. In your implementation, they go to quite different code paths.

```
// all primitive/varchar coercions are covered across row_to_row, list_to_list, and map_to_map 
"    row_to_row                 STRUCT<ti2si: TINYINT, si2int: SMALLINT, int2bi: INT, bi2vc: BIGINT>," +
"    list_to_list               ARRAY<STRUCT<ti2int: TINYINT, si2bi: SMALLINT, bi2vc: BIGINT>>," +
"    map_to_map                 MAP<TINYINT, STRUCT<ti2bi: TINYINT, int2bi: INT, float2double: " + floatToDoubleType + ">>" +